src/buffer/out/textBuffer.cpp

```diff
@@ -2353,3 +2354,86 @@ void TextBuffer::CopyHyperlinkMaps(const TextBuffer& other)
     _hyperlinkMap = other._hyperlinkMap;
     _hyperlinkCustomIdMap = other._hyperlinkCustomIdMap;
 }
+
+// Method Description:
+// - Adds a regex pattern we should search for
+// - The searching does not happen here, we only search when asked to by TerminalCore
+// Arguments:
+// - The regex pattern
+// Return value:
+// - An ID that the caller should associate with the given pattern
+const size_t TextBuffer::AddPatternRecognizer(const std::string regexString)
+{
+    ++_currentPatternId;
+    _IdsAndPatterns.emplace(std::make_pair(_currentPatternId, regexString));
+    return _currentPatternId;
+}
+
+// Method Description:
+// - Finds patterns within the requested region of the text buffer
+// Arguments:
+// - The firstRow to start searching from
+// - The lastRow to search
+// Return value:
+// - A vector containing 3-element tuples where
+//      tuple<0>: ID of the pattern
+//      tuple<1>: start coordinate of the pattern
+//      tuple<2>: end coordinate of the pattern
+const std::vector<std::tuple<size_t, COORD, COORD>> TextBuffer::UpdatePatterns(const size_t firstRow, const size_t lastRow) const
+{
+    std::vector<std::tuple<size_t, COORD, COORD>> result;
+
+    std::wstring concatAll;
+    const auto rowSize = GetRowByOffset(0).size();
+    concatAll.reserve(rowSize * (lastRow - firstRow + 1));
+
+    // to deal with text that spans multiple lines, we will first concatenate
+    // all the text into one string and find the pattern in that string
+    // later, we convert the locations of the patterns back into (row, col) coords
+    for (auto i = firstRow; i <= lastRow; ++i)
+    {
+        auto row = GetRowByOffset(i);
+        concatAll += row.GetCharRow().GetText();
+    }
+
+    // convert the string into something the regex iterator can work with
+    const auto constAll = til::u16u8(concatAll);

```

Comment:
It does look like there's a `wregex`. It's probably more efficient to convert the search string once on the way in when it's registered to `wchar_t` than it is to keep converting pieces of the buffer over and over just so the regex engine can match it.